go/integTest/glide_test_suite_test.go

```diff
@@ -108,13 +138,66 @@ func runClusterManager(suite *GlideTestSuite, args []string, ignoreExitCode bool
 	return string(output)
 }
 
-func extractServerVersion(output string) string {
-	// Redis response:
-	// Redis server v=7.2.3 sha=00000000:0 malloc=jemalloc-5.3.0 bits=64 build=7504b1fedf883f2
-	// Valkey response:
-	// Server v=7.2.5 sha=26388270:0 malloc=jemalloc-5.3.0 bits=64 build=ea40bb1576e402d6
-	versionSection := strings.Split(output, "v=")[1]
-	return strings.Split(versionSection, " ")[0]
+func getServerVersion(suite *GlideTestSuite) string {
+	var err error = nil
+	if len(suite.standaloneHosts) > 0 {
+		config := api.NewGlideClientConfiguration().
+			WithAddress(&suite.standaloneHosts[0]).
+			WithUseTLS(suite.tls).
+			WithRequestTimeout(5000)
+
+		client, err := api.NewGlideClient(config)
+		if err == nil && client != nil {
+			defer client.Close()
+			// TODO use info command
+			info, _ := client.CustomCommand([]string{"info", "server"})
+			return extractServerVersion(suite, info.(string))
+		}
+	}
+	if len(suite.clusterHosts) == 0 {
+		if err != nil {
```

Comment:
Sorry for the late comment, but IDE is warning this `err` will always be `nil`, which make sense, as all assignment of `err` is in the previous if clause.

Reply:
I think it could come non-`nil` from line 150

Reply:
Ah, I see, if both `len(suite.standaloneHosts) > 0` and `len(suite.clusterHosts) == 0` are true, and the call to `api.NewGlideClient(config)` did return an error, then it is possible that `err != nil` could be true here.

Linter problem, ignored.